redash/query_runner/athena.py

```diff
@@ -1,102 +1,127 @@
+                    'type': 'string',
+                    'title': 'KMS Key',
+                },
             },
-            'required': ['region', 'aws_access_key', 'aws_secret_key', 's3_staging_dir'],
-            'secret': ['aws_secret_key']
+            'required': ['region_name', 'schema_name', 's3_staging_dir']
         }
 
     @classmethod
-    def annotate_query(cls):
-        return ANNOTATE_QUERY
+    def enabled(cls):
+        return enabled
+
+    @classmethod
+    def type(cls):
+        return "athena"
+
+    def __init__(self, configuration):
+        super(Athena, self).__init__(configuration)
 
     def get_schema(self, get_stats=False):
         schema = {}
         query = """
         SELECT table_schema, table_name, column_name
         FROM information_schema.columns
-        WHERE table_schema NOT IN ('pg_catalog', 'information_schema')
+        WHERE table_schema NOT IN ('information_schema')
         """
 
         results, error = self.run_query(query, None)
-
         if error is not None:
             raise Exception("Failed getting schema.")
 
         results = json.loads(results)
-
         for row in results['rows']:
-            table_name = '{}.{}'.format(row['table_schema'], row['table_name'])
-
+            table_name = '{0}.{1}'.format(row['table_schema'], row['table_name'])
             if table_name not in schema:
                 schema[table_name] = {'name': table_name, 'columns': []}
-
             schema[table_name]['columns'].append(row['column_name'])
 
         return schema.values()
 
     def run_query(self, query, user):
+        cursor = pyathena.connect(**self.configuration.to_dict()).cursor()
+
         try:
-            data = {
-                'athenaUrl': 'jdbc:awsathena://athena.{}.amazonaws.com:443/'.format(self.configuration['region'].lower()),
-                'awsAccessKey': self.configuration['aws_access_key'],
-                'awsSecretKey': self.configuration['aws_secret_key'],
-                's3StagingDir': self.configuration['s3_staging_dir'],
-                'query': query
-            }
-
-            response = requests.post(PROXY_URL, json=data)
-            response.raise_for_status()
-
-            json_data = response.content.strip()
+            cursor.execute(query)
+            column_tuples = [(i[0], _TYPE_MAPPINGS.get(i[1], None)) for i in cursor.description]
+            columns = self.fetch_columns(column_tuples)
+            rows = [dict(zip(([c['name'] for c in columns]), r)) for i, r in enumerate(cursor.fetchall())]
+            data = {'columns': columns, 'rows': rows}
+            json_data = json.dumps(data, cls=JSONEncoder)
             error = None
-
-            return json_data, error
-        except requests.RequestException as e:
-            if e.response.status_code == 400:
-                return None, response.content
-
-            return None, str(e)
-        except KeyboardInterrupt:
-            error = "Query cancelled by user."
```

Comment:
Let's keep the special handler for `KeyboardInterrupt` to have the friendlier error message.